src/odemis/acq/fastem.py

```diff
@@ -167,6 +167,75 @@ def __init__(self, name, coordinates):
         # if None -> acquire, if not None, don't acquire again
 
 
+SCANNER_CONFIG_OVERVIEW = {
+    "multibeam_mode": False,
+    "external": False,
+    "blanker": None,  # automatic
+    "immersion": False,  # disable to get a larger field of view
+    "horizontal_fov": 1.5e-3,
+    "resolution": (1024, 884),  # px
+}
+
+
+SCANNER_CONFIG_MEGAFIELD = {
+    "multibeam_mode": True,
+    "external": False,
+    "blanker": False,
+    "immersion": True,
+    "horizontal_fov": 2.2e-5,
+    "resolution": (1024, 884),
+}
+
+
+SCANNER_CONFIG_LIVESTREAM = {
+    "multibeam_mode": False,
+    "external": False,
+    "blanker": None,  # automatic
+    "immersion": True,
+    "horizontal_fov": 2.2e-5,
+    "resolution": (1024, 884),
+}
+
+
+def _configure_scanner(scanner, multibeam_mode, external, blanker, immersion, horizontal_fov, resolution):
+    """
+    Configure the scanner by setting the VAs in the right order.
+    :param scanner (model.Emitter):
+    :param multibeam_mode (bool):
+    :param external (bool):
+    :param blanker (bool or None): None for automatic blanking at the end of the acquisition or when pausing the stream
+    :param immersion (bool):
+    :param horizontal_fov (float):
+    :param resolution (int, int):
+    """
+
+    scanner.multiBeamMode.value = multibeam_mode
+    scanner.external.value = external
+    scanner.blanker.value = blanker
+    # Immersion needs to be set before changing the horizontalFoV, as the range is updated
+    scanner.immersion.value = immersion
+
+    scanner.horizontalFoV.value = horizontal_fov  # m
+
+    # => compute the scale needed in X, use the same one in Y, and then compute the Y resolution.
+    # => set resolution to full FoV, and then adjust
+    scale = scanner.shape[0] / resolution[0]
+    scanner.scale.value = (scale, scale)
+    if scanner.resolution.value != TILE_RES:
+        logging.warning("Unexpected resolution %s on e-beam scanner, expected %s",
+                        scanner.resolution.value, TILE_RES)
+
+    md = scanner.getMetadata()
```

Comment:
How does this work now. You deleted below the method that reads the rotation for multibeam acquisition on the scanner from the MD. Here, you now always set the single beam rotation right? Do you not need to pick the right MD depending on the mode (live, megafield acq or overview acq)?

Reply:
I don't know. Is this something that is different for each mode? If so, how?